pkg/agent/datamodel/types.go

```diff
@@ -2235,11 +2235,25 @@ type PrivateEgress struct {
 	ProxyAddress            string `json:"proxyAddress"`
 }
 
+func (s *SecurityProfile) EnableBootstrapProfileContainerRegistry() bool {
```

Comment:
>func [](http://example.com/codeflow?start=0&length=4)

1. can we safe one function?
just check whether server name equals empty in yml template?
Only write that file if server name not empty, which makes sense.


2. organize the names?
assume PrivateEgress == BootstrapProfile

in (*SecurityProfile), just align to property
GetPrivateEgressContainerRegistryServer


In baker function, align it with real api.
EnableBootstrapProfileArtifactSourceCache

Then in future, we might change privateegress to bootstrapprofile without changing baker function.

Reply:
> can we safe one function?
just check whether server name equals empty in yml template?
Only write that file if server name not empty, which makes sense.

Fixed.

```
"EnableBootstrapProfileArtifactSourceCache": func() bool {
			return config.ContainerService.Properties.SecurityProfile.EnablePrivateEgress() && config.ContainerService.Properties.SecurityProfile.GetPrivateEgressContainerRegistryServer() != ""
		},
```

Reply:
> organize the names?

fixed. But in future, we still need to update baker function when changing privateegress to bootstrapprofile.

```
		"GetBootstrapProfileContainerRegistryServer": func() string {
			return config.ContainerService.Properties.SecurityProfile.GetPrivateEgressContainerRegistryServer()
		},
		"EnableBootstrapProfileArtifactSourceCache": func() bool {
			return config.ContainerService.Properties.SecurityProfile.EnablePrivateEgress() && config.ContainerService.Properties.SecurityProfile.GetPrivateEgressContainerRegistryServer() != ""
		},
```

Reply:
no template change then.